### ISSUE TITLE
carbonserver known metrics bug?

### ISSUE THREAD

**Issue by tehlers320** [go-graphite/go-carbon#146]
Im not sure if its a bug or how this is suppose to function but the known metrics seems to go up and then drop right off again.

![screen shot 2017-03-10 at 12 14 20 pm](https://cloud.githubusercontent.com/assets/16598119/23807432/44b794b4-058b-11e7-9d15-15293481f2b8.png)

I also have 0 cache hits/misses, so i've probably done something wrong.

**Comment by Civil** [comment]
What's the carbonserver config section?

**Comment by tehlers320** [comment]
@Civil stock config from the example, sorry i should know better than to not post a config.

Im on go-carbon 0.9.1 but the same behavior was exhibited in 0.9.0

```
[carbonserver]
listen = ":8080"
# Carbonserver support is still experimental and may contain bugs
# Or be incompatible with github.com/grobian/carbonserver
enabled = true
# Buckets to track response times
buckets = 10
# Maximum amount of globs in a single metric
max-globs = 100
# carbonserver-specific metrics will be sent as counters
# For compatibility with grobian/carbonserver
metrics-as-counters = false
# Read and Write timeouts for HTTP server
read-timeout = "60s"
write-timeout = "60s"
# carbonserver keeps track of all available whisper files
# in memory. This determines how often it will check FS
# for new metrics.
scan-frequency = "5m0s"
```

**Comment by Civil** [comment]
> enabled = false

That's in stock, but you seems to use it. So I'm mostly sure it's not a stock config :)

**Comment by tehlers320** [comment]
@Civil copy paste error. enabled = true i set.

**Comment by Civil** [comment]
And what kind of storage do you have?

**Comment by tehlers320** [comment]
SSDs, i2.4xlarge 4x800 SSD in raid 10

What does that have to do with the metrics stored? The file scan takes too long?

**Comment by tehlers320** [comment]
Have the same thing with i3.4xlarges (the new AWS instances that use NVMe drives)

**Comment by Civil** [comment]
> What does that have to do with the metrics stored? The file scan takes too long?

I just thought that may be there is some problem with that. Now I've looked at the graph and read the code and I think I know what's the bug is. It's about how carbonserver treats it's internal metrics. Honestly I've used it only with "metrics-as-counters = true", and forgot to check how this metric sent out if it's treated as gauge. That's the problem with this particular counter and I think I'll fix it soon (and will also check other internal metrics of carbonserver).
